tt_metal/hw/inc/dev_msgs.h

```diff
@@ -67,6 +67,15 @@ enum dispatch_core_processor_masks {
     DISPATCH_CLASS_MASK_ETH_DM0 = 1 << DISPATCH_CLASS_ETH_DM0,
 };
 
+enum core_type {
```

Comment:
The (new) HAL in core_config.h contains this sort of info per device.  This enum could be AddressableCoreType

Reply:
Added this type in latest commit